staging/src/k8s.io/apiserver/pkg/storage/etcd3/errors.go

```diff
@@ -30,13 +32,26 @@ func interpretWatchError(err error) error {
 	return err
 }
 
-func interpretListError(err error, paging bool) error {
-	switch {
-	case err == etcdrpc.ErrCompacted:
-		if paging {
-			return errors.NewResourceExpired("The provided from parameter is too old to display a consistent list result. You must start a new list without the from.")
-		}
+func interpretListError(err error, continueKey, keyPrefix string) error {
```

Comment:
I don't like changing the signature of this method to key off of "continueKey" present.  This method above all needs to be easy to read, even if you have to duplicate messages (put the string in a constant).